src/main/java/org/jenkinsci/bytecode/Transformer.java

```diff
@@ -57,16 +58,28 @@ public synchronized void loadRules(Collection<? extends ClassLoader> loaders) th
             return image;
 
         final ClassReader cr = new ClassReader(image);
-        final ClassWriter cw = new ClassWriter(/*ClassWriter.COMPUTE_FRAMES|*/ClassWriter.COMPUTE_MAXS);
+        final ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES|ClassWriter.COMPUTE_MAXS);
 
         final boolean[] modified = new boolean[1];
 
-        cr.accept(new ClassVisitor(ASM5,cw) {
+        // If code contains JSR/RET instructions then ASM fails to transform it with
+        // java.lang.RuntimeException: JSR/RET are not supported with computeFrames option
+        // so inline any JSR subroutines
```

Comment:
OK…just bear in mind that a lot of plugin releases are still compiled with `-target 5`.